index.js

```diff
@@ -1,117 +1,142 @@
 const express = require("express");
+const morgan = require("morgan");
+const bodyParser = require("body-parser");
+const uuid = require("uuid");
+
 const app = express();
-const morgan = require("morgan")
-bodyParser = require('body-parser'),
-      uuid = require('uuid');
-
-    const app = express();
-    app.use(bodyParser.json());
-
-    let topMovies = [
-{
- id: 1,
- title: 'harry potter and the sorcerer\'s stone',
- author: 'J.K. Rowling',
- director: 'this person',
- genre: 'this genre'
-},
-{
- id: 2,
- title: 'Lord of the rings',
- author:'J.R.R. Tolkien',
- director: 'this person',
- genre: 'this genre'
-},
-{
- id: 3,
- title: 'Twilight',
- author: 'stephanie Meyer',
- director: 'this person',
- genre: 'this genre'
- }
-];
+app.use(morgan("common"));
+app.use(bodyParser.json());
 
-app.use(morgan("common")); 
 app.use(express.static("public"));
 
-// gets the data about a single director, by name \\
+let topMovies = [
+  {
+    id: 1,
+    title: "harry potter and the sorcerer's stone",
+    author: "J.K. Rowling",
+    director: "this person",
+    genre: "this genre",
+  },
+  {
+    id: 2,
+    title: "Lord of the rings",
+    author: "J.R.R. Tolkien",
+    director: "this person",
+    genre: "this genre",
+  },
+  {
+    id: 3,
+    title: "Twilight",
+    author: "stephanie Meyer",
+    director: "this person",
+    genre: "this genre",
+  },
+];
 
-app.get('/movies/:director', (req, res) => {
-  res.json(topMovies.find((movies) => {
-  return movies.director ===req.params.director
-  }));
-  });
-  
-  app.get('/movies/:director/:genre', (req, res) => {
-    res.json(topMovies.find((movies) => {
-    return movies.genre ===req.params.genre
-    }));
-    });
-  
-  //adds data for a new movie to our list of movies. \\
-  
-  app.post('/movies', (req, res) => {
+app.get("/movies", (req, res) => {
+  res.json(topMovies);
+});
+
+//Get a single movie by title
+app.get("/movies/:title", (req, res) => {
+  res.json(
+    topMovies.find((movies) => {
+      return movies.title === req.params.title;
+    })
+  );
+});
+
+//Get list of movies by director
+app.get("/movies/director/:director", (req, res) => {
+  res.json(
+    topMovies.filter((movies) => {
+      return movies.director === req.params.director;
+    })
+  );
+});
```

Comment:
This is one of the endpoints that was updated to avoid a collision of paths. You'll notice that the word director is repeated twice, but one of them has a colon. The part that has a colon is the argument and it really it can be named anything. For example `app.get("/movies/director/:name`

```
app.get("/movies/director/:name", (req, res) => {
    res.json(
      topMovies.filter((movies) => {
        return movies.director === req.params.name;
      })
    );
  });

```
The word `director` that doesn't have a colon is actually part of the URL. So when you use the endpoint it will look something like this `/movies/director/Steven Allan Spielberg`. The part that has the colon is "dynamic"